### PR TITLE
Do not apply mup initialization to parameters of LayerNorm and BatchNorm layers

### PR DESCRIPTION
**Pull request by rao-ma**
…orm layers

### CODE REVIEW

File: scripts/run.py [hunk 1]
```diff
@@ -154,6 +154,23 @@ def main(config: ConfigBase):
     # Initialise the model with mup
     named_params = list(model.named_parameters())
     param_names = {name for name, _ in named_params}
+
+    modules_without_init = set()
+    for module_name, module_type in model.named_modules():
+        if isinstance(module_type, torch.nn.LayerNorm) or isinstance(module_type, torch.nn.BatchNorm1d) or isinstance(module_type, torch.nn.BatchNorm2d) or isinstance(module_type, torch.nn.BatchNorm3d):
```
Comment (runame) [inline scripts/run.py]: Here you can use `torch.nn.modules.batchnorm._BatchNorm` instead of all the BN subclasses.

File: scripts/run.py [hunk 2]
```diff
@@ -154,6 +154,23 @@ def main(config: ConfigBase):
...
+            logging.info(f"Module without mup initialization: {module_name} {module_type}")
+            modules_without_init.add(module_name)
+    params_without_init = set()
+    for param_name in param_names:
+        for module_name in modules_without_init:
+            if param_name.rsplit('.', 1)[0] == module_name:
+                params_without_init.add(param_name)
```
Comment (BrunoKM) [inline scripts/run.py]: Might be a bit safer/shorter:
```python
params_without_init = set()
for module_name, module_type in model.named_modules():
        if isinstance(module_type, torch.nn.LayerNorm) or isinstance(module_type, torch.nn.BatchNorm1d) or isinstance(module_type, torch.nn.BatchNorm2d) or isinstance(module_type, torch.nn.BatchNorm3d):
            logging.info(f"Module without mup initialization: {module_name} {module_type}")
            params_without_init.update({get_param_name(model, param) for param in module_type.parameters()})
```

File: scripts/run.py [hunk 3]
```diff
@@ -154,6 +154,23 @@ def main(config: ConfigBase):
     # Initialise the model with mup
     named_params = list(model.named_parameters())
     param_names = {name for name, _ in named_params}
+
+    modules_without_init = set()
```
Comment (runame) [inline scripts/run.py]: I think it probably makes sense to move this whole code block in a separate function?

File: scripts/run.py [hunk 4]
```diff
@@ -176,6 +193,7 @@ def main(config: ConfigBase):
             param_inf_types=param_inf_types,
             init_scale=init_scales,
             distribution=config.initialisation.init_distribution,
+            params_without_init=params_without_init
```
Comment (BrunoKM) [inline scripts/run.py]: I think we also need the same rule for `sp_initialise`, `pytorch_initialise`
